mne/source_space.py

```diff
@@ -2030,6 +2038,9 @@ def _filter_source_spaces(surf, limit, mri_head_t, src, n_jobs=1,
             extras += [limit]
             logger.info('%d source space point%s omitted because of the '
                         '%6.1f-mm distance limit.' % tuple(extras))
+        # Adjust the patch inds as well if necessary
+        if s.get('patch_inds') is not None:
+            s['patch_inds'] = s['patch_inds'][~outside]
```

Comment:
This is the fix -- the `patch_inds` are used to adjust the surface normals, and if points are omitted/filtered out during the forward solution calculation stage (by distance from or being outside the inner skull surface) then these would be off by some indices.